scene/2d/polygon_2d.cpp

```diff
@@ -364,7 +368,31 @@ void Polygon2D::_notification(int p_what) {
 
 				arr[RS::ARRAY_INDEX] = index_array;
 
-				RS::get_singleton()->mesh_add_surface_from_arrays(mesh, RS::PRIMITIVE_TRIANGLES, arr, Array(), Dictionary(), RS::ARRAY_FLAG_USE_2D_VERTICES);
+				RS::SurfaceData sd;
+
+				if (skeleton_node) {
+					// Compute transform between mesh and skeleton for runtime AABB compute.
+
+					const Transform2D meshTransform = this->get_global_transform();
+					const Transform2D skeletonTransform = skeleton_node->get_global_transform();
+					const Transform2D mesh_to_sk2d = meshTransform * skeletonTransform.inverse();
```

Comment:
It may be better to use `affine_inverse()` here (that's what I used in 3.x). `inverse()` just uses the transpose which is cheaper but doesn't work in some situations (I think transpose depends on being orthonormal but I forget).

Reply:
Ya, transpose is only valid if the matrix is orthonormal. I don't know if this transform is guaranteed to be orthonormal or not. So it's probably safest to use `affine_inverse()`